specification/quantum/data-plane/Microsoft.Quantum/preview/2019-11-04-preview/quantum.json

```diff
@@ -0,0 +1,552 @@
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/jobIdParameter"
+          }
+        ],
+        "x-ms-examples": {
+          "Get a job's metadata": {
+            "$ref": "./examples/jobGet.json"
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/jobDetails"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/restError"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Jobs_Put",
+        "description": "Create a job.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/jobIdParameter"
+          },
+          {
+            "name": "jobDefinition",
+            "description": "The complete metadata of the job to submit.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/jobDetails"
+            }
+          }
+        ],
+        "x-ms-examples": {
+          "Submit a new job for execution to a Quantum Workspace": {
+            "$ref": "./examples/jobPut.json"
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/jobDetails"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/jobDetails"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/restError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "Jobs_Delete",
+        "description": "Cancel a job.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/jobIdParameter"
+          }
+        ],
+        "x-ms-examples": {
+          "Cancel the execution of an existing job": {
+            "$ref": "./examples/jobDelete.json"
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/jobDetails"
```

Comment:
Why would a DELETE return the resource? Is that expected?

Reply:
Or, related, is there a difference that a client can understand between receiving a 200 and a 204 response? E.g. are you following the ARM convention of returning 200 if the resource was actually deleted and 204 if the request was well-formed, but not such resource existed?

Reply:
No, we're just returning 204 if the job is succesfully cancelled.